pkg/storage/action_cache_asset_store.go

```diff
@@ -34,7 +34,11 @@ func NewActionCacheAssetStore(actionCache, contentAddressableStorage blobstore.B
 }
 
 func (rs *actionCacheAssetStore) assetToDirectory(ctx context.Context, asset *asset.Asset, instance digest.InstanceName) (*remoteexecution.Directory, error) {
-	digest, err := instance.NewDigestFromProto(asset.Digest)
+	digestFunction, err := instance.GetDigestFunction(remoteexecution.DigestFunction_UNKNOWN, len(asset.GetDigest().GetHash()))
```

Comment:
Yeah, this is the best that you can do right now, given that the Remote Asset protocol doesn't take the digest function explicitly.

I think this method is never called with `asset == nil`, right? If that's the case, then be sure to use `asset.Digest.GetHash()`. Then it's (implicitly) self-documenting that this holds.

Reply:
Done